OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -160,7 +160,7 @@ public void ScanAndAttack(Actor self, bool allowMove)
 		void Attack(Actor self, Actor targetActor, bool allowMove)
 		{
 			TargetedActor = targetActor;
-			var target = Target.FromActor(targetActor);
+			var target = Target.FromActor(targetActor, TargetMethod.Automatic);
```

Comment:
Food for thought for reviewers: this is called both from `ScanAndAttack()` (in which case `Automatic` is perfectly fine) and `Damaged()`, in which case ...
1) If we set it as Automatic, that means the actor may end up in a situation where it doesn't return fire.
2) If we set it as Manual ... well that may lead to confusion and potentially bad handling by future code, because the player did *not* order the attack, it was an automated response.
3) We could introduce a third value for `TargetMethod`, but that may end up being completely redundant (same as `Manual`, just with a more appropriate name for this one case).